dump_manager_bmc.cpp

```diff
@@ -101,33 +119,35 @@ uint32_t Manager::captureDump(Type type,
 
         // get dreport type map entry
         auto tempType = TypeMap.find(type);
-
         execl("/usr/bin/dreport", "dreport", "-d", dumpPath.c_str(), "-i",
               id.c_str(), "-s", std::to_string(size).c_str(), "-q", "-v", "-p",
               fullPaths.empty() ? "" : fullPaths.front().c_str(), "-t",
               tempType->second.c_str(), nullptr);
 
         // dreport script execution is failed.
         auto error = errno;
-        log<level::ERR>(
-            fmt::format(
-                "Error occurred during dreport function execution, errno({})",
-                error)
-                .c_str());
+        log<level::ERR>(fmt::format("Error occurred during dreport "
+                                    "function execution, errno({})",
+                                    error)
+                            .c_str());
         elog<InternalFailure>();
     }
     else if (pid > 0)
     {
-        auto rc = sd_event_add_child(eventLoop.get(), nullptr, pid,
-                                     WEXITED | WSTOPPED, callback, nullptr);
+        // local variable goes out of scope using pointer, callback method
+        // need to dellocate the pointer
+        Type* typePtr = new Type();
```

Comment:
C style coding clearly here. Use CPP style approach either by using std unique pointer and make_unique. Also use static_cast for casting to void pointer